console/collections/src/kary_merkle_tree/helpers/path_hash.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2019-2023 Aleo Systems Inc.
+// This file is part of the snarkVM library.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::BooleanHash;
+use snarkvm_console_algorithms::{Keccak, Poseidon, BHP};
+use snarkvm_console_types::prelude::*;
+
+#[cfg(not(feature = "serial"))]
+use rayon::prelude::*;
+
+/// A trait for a Merkle path hash function.
+pub trait PathHash: Clone + Send + Sync {
+    type Hash: Copy + Clone + Debug + Default + PartialEq + Eq + FromBytes + ToBytes + Send + Sync;
+
+    /// Returns the hash of the given child nodes.
+    fn hash_children(&self, children: &[Self::Hash]) -> Result<Self::Hash>;
+
+    /// Returns the empty hash.
+    fn hash_empty<const ARITY: u8>(&self) -> Result<Self::Hash> {
+        let children = (0..ARITY).map(|_| Self::Hash::default()).collect::<Vec<_>>();
+        self.hash_children(&children)
+    }
+
+    /// Returns the hash for each tuple of child nodes.
+    fn hash_all_children(&self, child_nodes: &[Vec<Self::Hash>]) -> Result<Vec<Self::Hash>> {
+        match child_nodes.len() {
+            0 => Ok(vec![]),
+            1..=100 => child_nodes.iter().map(|children| self.hash_children(children)).collect(),
```

Comment:
same note on the choice of `100` as before

Reply:
I think it remains the same given 2-ary is also supported here.